### ISSUE TITLE
Range header field ABNF

### ISSUE THREAD

**Issue by reschke** [httpwg/http-core#196]
The Range header field ABNF:
~~~
  Range = byte-ranges-specifier / other-ranges-specifier
  other-ranges-specifier = other-range-unit "=" other-range-set
  other-range-set = 1*VCHAR
~~~
This is another misguided attempt to restrict syntax using ABNF in presence of extension points.

For instance,
~~~
bytes=abc
~~~

matches `other-ranges-specifier`.

The proper fix for this is to rewrite the ABNF so it's properly layered - a generic one, plus an overlay for the only range unit we define.

**Comment by royfielding** [comment]
I am very far down the rabbit hole on this one.

**Comment by royfielding** [comment]
Fixed by 7594239b46ef2c755b2e91a4572d86707d5f1021
